crates/btcio/src/reader/query.rs

```diff
@@ -229,11 +305,28 @@ async fn fetch_and_process_block<R: Reader>(
     status_updates: &mut Vec<L1StatusUpdate>,
 ) -> anyhow::Result<BlockHash> {
     let block = ctx.client.get_block_at(height).await?;
+    process_block(ctx, state, status_updates, height, block).await
+}
+
+async fn process_block<R: Reader>(
+    ctx: &ReaderContext<R>,
+    state: &mut ReaderState,
+    status_updates: &mut Vec<L1StatusUpdate>,
+    height: u64,
+    block: Block,
+) -> anyhow::Result<BlockHash> {
     let txs = block.txdata.len();
 
     let params = ctx.config.params.clone();
-    let filter_config = TxFilterConfig::derive_from(params.rollup())?;
-    let filtered_txs = filter_protocol_op_tx_refs(&block, filter_config);
+    let filtered_txs = match state.filter_config() {
+        EpochFilterConfig::AtEpoch(config) => filter_protocol_op_tx_refs(&block, config),
+        EpochFilterConfig::AtTransition(from_config, to_config) => {
+            let mut txs = filter_protocol_op_tx_refs(&block, from_config);
+            let txs1 = filter_protocol_op_tx_refs(&block, to_config);
+            txs.extend_from_slice(&txs1);
+            txs
+        }
```

Comment:
Why are we looking at both epochs here?  We only care about a single epoch at a time.

If the same tx matched under both sets of rules, then the entry would be present in the list twice, which doesn't make sense.

Reply:
Oh yes, this is kind of doesn't make sense right now. We can uniquely identify protocol ops, how I haven't thought of yet. This would be unnecessary if there were no transitions, but transitions seem inevitable don't they?